Larionov_Alex_labwork

Alexej-Larionov opened the pull request:
Sorry! Work is still in progress. But I at least managed to create my first ever pull request.

Alexej-Larionov committed:
Larionov_Alex_labwork

Alexej-Larionov committed:
Update Поразрядная сортировка.cpp

ValentinV95 reviewed (commented):
Такая поразрядная сортировка не будет работать для типа данных double.
Не во всех сортировках есть подсчет сравнений и перестановок
Не нашел отчета и главной программы, которую можно запускать

Alexej-Larionov committed:
Update Поразрядная сортировка.cpp

Alexej-Larionov committed:
Add files via upload

Почему-то не заполняется массив возможно что-то не так со switch?

Alexej-Larionov committed:
Add files via upload

Исправил, завтра отправлю отчет, извините за задержку.

Alexej-Larionov committed:
Add files via upload

Alexej-Larionov committed:
Add files via upload

Alexej-Larionov committed:
Delete Отчет.docx

ValentinV95 commented on Larionov_Alex/Lab_sorts.h:
```diff
@@ -0,0 +1,152 @@
+	{
+		array[N - k + i] = additional_array[i]; a++;
+	}
+	memcpy(array, additional_array, N * sizeof(double));
```
Данная строка отменяет все изменения предыдущих двух циклов, подобную ошибку я уже видел у ваших одногруппников

ValentinV95 reviewed (commented):
Есть замечание по коду, так же необходимо удалить дубликаты файлов с сортировками

Alexej-Larionov committed:
Update Lab_sorts.h

Убрал.

Alexej-Larionov committed:
Delete Insertion_sort.cpp

Alexej-Larionov committed:
Delete Shell's_sort.cpp

Alexej-Larionov committed:
Delete Поразрядная сортировка.cpp

Alexej-Larionov committed:
Delete Сортировка_слиянием.cpp

Alexej-Larionov commented:
Удалил ненужные файлы и лишние строки поразрядной сортировки.

ValentinV95 commented on Larionov_Alex/Lab_sorts.h:
```diff
@@ -0,0 +1,145 @@
+	{
+		array[i] = additional_array[N - 1 - i]; a++;
+	}
+	printf("\n%d%s\n%d%s\n", a, "swaps", b, "comparisons");
```
Перед сдачей работ проверяете работоспособность сортировки. Сейчас она у вас некорректно обрабатывает числа с плавающий запятой

ValentinV95 reviewed (commented):
Замечания по коду исправлены неверно. Также появилась утечка памяти

Alexej-Larionov committed:
Delete Final_Lab.exe

Alexej-Larionov committed:
Delete Final_Lab.cpp

Alexej-Larionov committed:
Delete Lab_sorts.h

Alexej-Larionov committed:
Add files via upload

Поправил и проверил поразрядную сортировку, дополнил функцию создания массива опцией введения массива вручную, изменил первую опцию заполнения массива(теперь генерируются числа разные по знаку, целой и дробной частям), совершил мелкие косметические изменения в коде.

ValentinV95 commented on Larionov_Alex/Lab_sorts.h:
```diff
@@ -0,0 +1,148 @@
+		{
+		array[N - k + i] = additional_array[i]; swaps++;
+		}
+	printf("swaps = %d  comparisons = % d\n", swaps,  comps);
```
Утечка памяти, в данной функции выделяются массивы  additional_array и counters, но они не удаляются

ValentinV95 reviewed (commented):
Осталось замечание по коду.
Отчет: в разделе результаты у вас сейчас только графики. Необходимо в этот раздел добавить некоторое словесное описание, полученных графиков и выводов по ним

Alexej-Larionov committed:
Update Lab_sorts.h

ValentinV95 commented:
Код принят

Alexej-Larionov committed:
Update Final_Lab.cpp

Alexej-Larionov committed:
Delete Отчет.docx

Alexej-Larionov committed:
Add files via upload

ValentinV95 commented:
Отчет принят
